src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -293,24 +308,66 @@ public TopicPartitionChannel(
         new SnowflakeTelemetryChannelStatus(
             tableName,
             connectorName,
-            channelName,
+            channelNameFormatV1,
             startTime,
             enableCustomJMXMonitoring,
             metricsJmxReporter,
             this.offsetPersistedInSnowflake,
             this.processedOffset,
             this.latestConsumerOffset);
     this.telemetryServiceV2.reportKafkaPartitionStart(
-        new SnowflakeTelemetryChannelCreation(this.tableName, this.channelName, startTime));
+        new SnowflakeTelemetryChannelCreation(this.tableName, this.channelNameFormatV1, startTime));
 
     if (lastCommittedOffsetToken != NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE) {
       this.sinkTaskContext.offset(this.topicPartition, lastCommittedOffsetToken + 1L);
     } else {
       LOGGER.info(
           "TopicPartitionChannel:{}, offset token is NULL, will rely on Kafka to send us the"
               + " correct offset instead",
-          this.getChannelName());
+          this.getChannelNameFormatV1());
+    }
+  }
+
+  /**
+   * Checks if the configuration provided in Snowflake Kafka Connect has set {@link
+   * SnowflakeSinkConnectorConfig#ENABLE_CHANNEL_OFFSET_TOKEN_MIGRATION_CONFIG} to any value. If not
+   * set, it fetches the default value.
+   *
+   * <p>If the returned is false, system function for channel offset migration will not be called
+   * and Channel name will use V1 format.
+   *
+   * @param sfConnectorConfig customer provided json config
+   * @return true is enabled, false otherwise
+   */
+  private boolean isEnableChannelOffsetMigration(Map<String, String> sfConnectorConfig) {
+    boolean isEnableChannelOffsetMigration =
+        Boolean.parseBoolean(
+            sfConnectorConfig.getOrDefault(
+                SnowflakeSinkConnectorConfig.ENABLE_CHANNEL_OFFSET_TOKEN_MIGRATION_CONFIG,
+                Boolean.toString(ENABLE_CHANNEL_OFFSET_TOKEN_MIGRATION_DEFAULT)));
+    if (!isEnableChannelOffsetMigration) {
+      LOGGER.info(
+          "Config:{} is disabled for connector:{}",
+          ENABLE_CHANNEL_OFFSET_TOKEN_MIGRATION_CONFIG,
+          conn.getConnectorName());
     }
```

Comment:
@sfc-gh-rcheng is making some changes in this area to log all non-sensitive configurations

Reply:
@sfc-gh-rcheng do you mind waiting for my change to go in first? I think it would be easier for you to rebase.. But let me know if its not possible..

Reply:
yeah no worries, lets go with this change first!